crates/core/src/formatter/errors/documentation/view.rs

```diff
@@ -0,0 +1,158 @@
+//! View components for displaying error documentation.
+//!
+//! This module provides various view wrappers for error documentation,
+//! enabling the customizable display of different parts of error information
+//! such as summary, causes, fixes, and detailed descriptions.
+
+use std::fmt::Write;
+
+use colored::Colorize as _;
+
+use crate::formatter::PrettyFmt;
+
+use super::{
+    format_description, format_fixes, format_likely_causes, format_references, format_summary,
+    AnvilErrorDocumentation,
+};
+
+/// Displays the complete documentation for an error.
+///
+/// This view wraps an error and provides a fully-formatted display of all
+/// available documentation, including summary, causes, fixes, references,
+/// and detailed description.
+pub struct FullDocumentation<'a, E>(pub &'a E)
```

Comment:
Re our discussion yesterday: this is not an approach I would've taken, but it makes sense, and combined with traits introduced, I'm fine with it.